pkg/operator/ceph/object/objectstore.go

```diff
@@ -166,19 +352,118 @@ func setMultisite(context *Context, serviceIP string, spec cephv1.ObjectStoreSpe
+	if zoneIsMaster {
+		// endpoints that are part of a master zone are supposed to be the endpoints for a zone group
+		_, err := RunAdminCommandNoMultisite(objContext, "zonegroup", "modify", realmArg, zoneGroupArg, endpointArg)
+		if err != nil {
+			return errors.Wrapf(err, "failed to add object store %q in rgw zone group %q", objContext.Name, objContext.ZoneGroup)
+		}
+		logger.Debugf("endpoints for zonegroup %q are now %q", objContext.ZoneGroup, zoneEndpoints)
+
+		// check if zone group is master only if zone is master for creating the system user
+		zoneGroupIsMaster, err = checkZoneGroupIsMaster(objContext)
+		if err != nil {
+			return errors.Wrapf(err, "failed to find out whether zone group %q in is the master zone group", objContext.ZoneGroup)
+		}
+	}
+	_, err = RunAdminCommandNoMultisite(objContext, "zone", "modify", realmArg, zoneGroupArg, zoneArg, endpointArg)
+	if err != nil {
+		return errors.Wrapf(err, "failed to add object store %q in rgw zone %q", objContext.Name, objContext.Zone)
+	}
+
+	// the period will help notify other zones of changes if there are multi-zones
+	_, err = RunAdminCommandNoMultisite(objContext, "period", "update", "--commit", realmArg, zoneGroupArg, zoneArg)
+	if err != nil {
+		return errors.Wrap(err, "failed to update period")
+	}
+	logger.Infof("added object store %q to realm %q, zonegroup %q, zone %q", objContext.Name, objContext.Realm, objContext.ZoneGroup, objContext.Zone)
+
+	// create system user for realm for master zone in master zonegorup for multisite scenario
+	if zoneIsMaster && zoneGroupIsMaster {
+		uid := objContext.Realm + "-system-user"
+		uidArg := fmt.Sprintf("--uid=%s", uid)
+		output, err := RunAdminCommandNoMultisite(objContext, "user", "info", uidArg)
+		if err != nil {
+			if code, ok := exec.ExitStatus(err); ok && code == int(syscall.EINVAL) {
+				logger.Debugf("realm system user %q not found, running `radosgw-admin user create`", uid)
+				accessKeyArg, secretKeyArg, err := GetRealmKeyArgs(objContext.Context, objContext.Realm, namespace)
+				if err != nil {
+					return errors.Wrap(err, "failed to get keys for realm")
+				}
+				systemArg := "--system"
+				displayNameArg := fmt.Sprintf("--display-name=%s.user", objContext.Realm)
+				output, err = RunAdminCommandNoMultisite(objContext, "user", "create", realmArg, zoneGroupArg, zoneArg, uidArg, displayNameArg, accessKeyArg, secretKeyArg, systemArg)
+				if err != nil {
+					return errors.Wrapf(err, "failed to create realm system user %q for reason: %q", uid, output)
+				}
+			} else {
+				return errors.Wrapf(err, "radosgw-admin user info failed with code %d and output %q", code, output)
+			}
+		} else {
+			logger.Debugf("realm system user %q has already been created", uid)
+		}
+	}
+
+	return nil
+}
+
+func setMultisite(objContext *Context, store *cephv1.CephObjectStore, serviceIP string) error {
+	serviceEndpoint := fmt.Sprintf("http://%s:%d", serviceIP, store.Spec.Gateway.Port)
+	if store.Spec.Gateway.SecurePort != 0 {
```

Comment:
@travisn @leseb is this the correct way to determine whether or not and endpoint should be https?

Reply:
LGTM